pkg/processor/reportprocessor.go

```diff
@@ -145,6 +156,80 @@ func (p *Processor) createUpdatePolicyReports(input chan types.ProcessorData, dy
 	}
 }
 
+func convertSevFromGovernance(policySev string) string {
+	sevMapping := map[string]interface{}{
+		"critical": "4",
+		"high": "3",
+		"medium": "2",
+		"low": "1",
+	}
+	if severity, ok := sevMapping[policySev]; ok {
+		return severity.(string)
+	}
+	return "0"
+}
+
+//getGovernanceResults creates a result object for each policy violation in the cluster
+func getGovernanceResults(dynamicClient dynamic.Interface, clusterInfo types.ManagedClusterInfo) []*v1alpha2.PolicyReportResult {
+	glog.V(2).Infof(
+		"Getting policy violations for cluster %s (%s)",
+		clusterInfo.Namespace,
+		clusterInfo.ClusterID,
+	)
+
+	res := dynamicClient.Resource(policyGvr).Namespace(clusterInfo.Namespace)
+	policyList, err := res.List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		glog.V(2).Infof(
+			"Error getting policy data for cluster %s (%s)",
+			clusterInfo.Namespace,
+			clusterInfo.ClusterID,
+		)
+		return []*v1alpha2.PolicyReportResult{}
+	}
+
+	var clusterViolations []*v1alpha2.PolicyReportResult
+	for _, plc := range policyList.Items {
+		md := plc.Object["metadata"].(map[string]interface{})
+		status := plc.Object["status"].(map[string]interface{})
+		details := status["details"].([]interface{})
+		if status["compliant"].(string) == "NonCompliant" {
+			for _, detail := range details {
+				if detail.(map[string]interface{})["compliant"] == "NonCompliant" {
+					templateMeta := detail.(map[string]interface{})["templateMeta"].(map[string]interface{})
+					historyItem := detail.(map[string]interface{})["history"].([]interface{})[0].(map[string]interface{})
+					annotations := md["annotations"].(map[string]interface{})
+					clusterViolations = append(clusterViolations, &v1alpha2.PolicyReportResult{
+						Policy:      md["name"].(string),
+						Description: historyItem["message"].(string),
+						Scored:      false,
+						Category:    annotations["policy.open-cluster-management.io/categories"].(string),
+						Timestamp:   metav1.Timestamp{Seconds: time.Now().Unix(), Nanos: int32(time.Now().UnixNano())},
```

Comment:
Is the current `Now` more useful than the timestamp in the history's `lastTimestamp`?

Reply:
@JustinKuli I just used `Now` in the `Timestamp` because that's what insights does for their results field, so i assumed the field was supposed to be when the policyReport object was updated